nixos/modules/virtualisation/xen-boot-builder.sh

```diff
@@ -0,0 +1,165 @@
+# This script is called by ./xen-dom0.nix to create the Xen boot entries.
+# shellcheck shell=bash
+
+# Handle input argument and exit if the flag is invalid. See virtualisation.xen.efi.bootBuilderFlag below.
+[[ $# -ne 1 ]] && echo -e "\e[1;31merror:\e[0m xenBootBuilder must be called with exactly one argument. See the \e[1;34mvirtualisation.xen.efi.bootBuilderFlag\e[0m option." && exit 1
+case "$1" in
+    "--quiet") true ;;
+    "--release" | "--report") echo -n "Installing Xen Hypervisor boot entries..." ;;
+    "--debug") echo -e "\e[1;34mxenBootBuilder:\e[0m called with the '$1' flag" ;;
+    *)
+        echo -e "\e[1;31merror:\e[0m xenBootBuilder was called with an invalid argument. See the \e[1;34mvirtualisation.xen.efi.bootBuilderFlag\e[0m option."
+        exit 2
+        ;;
+esac
+
+# Get the current Xen generations and store them in an array. This will be used
+# for displaying the diff later, if xenBootBuilder was called with `--report`.
+# We also delete the current Xen entries here, as they'll be rebuilt later if
+# the corresponding NixOS generation still exists.
+mapfile -t preGenerations < <(find "$efiMountPoint"/loader/entries -type f -name 'xen-*.conf' | sort -V | sed 's/\/loader\/entries\/nixos/\/loader\/entries\/xen/g' | sed 's/^.*\/xen/xen/g' | sed 's/.conf$//g')
```

Comment:
Nit: Would prefer any delimiter other than `/` here for readability, and the three `sed`s can be combined into one.